In [34]:
import pandas as pd
import numpy as np

In [35]:
data={'Area':[2600,3000,3200,3600,4000],
      'Bedrooms':[3,4,np.nan,3,5],
      'Age':[20,15,18,30,8],
      'Price':[550000,565000,610000,595000,760000]}
df = pd.DataFrame(data)
df

Area  Bedrooms  Age   Price
0  2600       3.0   20  550000
1  3000       4.0   15  565000
2  3200       NaN   18  610000
3  3600       3.0   30  595000
4  4000       5.0    8  760000

In [36]:
#replace the missing values
m = df['Bedrooms'].median()
n=round(m,0)
n
df['Bedrooms'].fillna(n-1, inplace=True)

In [37]:
df

Area  Bedrooms  Age   Price
0  2600       3.0   20  550000
1  3000       4.0   15  565000
2  3200       3.0   18  610000
3  3600       3.0   30  595000
4  4000       5.0    8  760000

In [53]:
from sklearn import linear_model
lm= linear_model.LinearRegression()
lm.fit(X, y)
a= lm.predict([[4000,4,30]])
b= lm.predict([[4000,4,3]])
print(lm.coef_)
print(lm.intercept_)
print(a)
print(b)

[   137.25 -26025.    -6825.  ]
383724.9999999998
[623875.]
[808150.]


1. Calculate the coefficient of each factor.  
Ans:coefficients:  
        Area: 122.1511  
        Bedrooms: 53204.439421  
        Age:1163.1720
            
3. Predict the price of a property which is in 4000 unit in area, 4 BHK and 30 years old.  
Ans:623875
4. Predict the price of a property which is in 4000 unit in area, 4 BHK and 3 years old.  
Ans:808150
5. Based on point 3 and 4 give your remark why there is a change in same size property rate, if any.  
Ans:There is a change in price for same size property because of factor "Age". New property will cost more than older property.